### ISSUE TITLE
ohai chef_packages doesn't return chef info without full path to ohai

### ISSUE THREAD

**Issue by eheydrick** [chef/ohai#1177]
## Description

`ohai chef_packages` doesn't return any info for `chef` if ohai isn't run with the full path or `/opt/chef/embedded/bin` isn't on the `$PATH`.  This seems like a regression as under Chef 12 and 13.0.113 I did not need to fully-qualify the path to ohai to get chef package info.

## Ohai Version

chef-client 13.0.113  / ohai 13.0.0 is the last version that worked. chef-client 13.1.31 / ohai 13.1.0 is the first version that has this problem. It's still a problem in chef-client 14.0.202 / ohai 14.0.31.

## Platform Version

Ubuntu 16.04

### chef-client 13.0.113 / ohai 13.0.0 does output chef info:

```
root@0d97386294be:/# chef-client --version
Chef: 13.0.113
root@0d97386294be:/# ohai --version
Ohai: 13.0.0
root@0d97386294be:/# ohai chef_packages
[2018-04-26T18:37:43+00:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
{
  "ohai": {
    "version": "13.0.0",
    "ohai_root": "/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/ohai-13.0.0/lib/ohai"
  },
  "chef": {
    "version": "13.0.113",
    "chef_root": "/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.0.113/lib"
  }
}
```

### chef-client 13.1.31 / ohai 13.1.0 does not output chef info:

```
root@0d97386294be:/# chef-client --version
Chef: 13.1.31
root@0d97386294be:/# ohai --version
Ohai: 13.1.0
root@0d97386294be:/# ohai chef_packages
[2018-04-26T18:43:25+00:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
{
  "ohai": {
    "version": "13.1.0",
    "ohai_root": "/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/ohai-13.1.0/lib/ohai"
  }
}
```

If you supply the full path to ohai it works: 

```
root@0d97386294be:/# /opt/chef/embedded/bin/ohai chef_packages
[2018-04-26T18:44:34+00:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
{
  "ohai": {
    "version": "13.1.0",
    "ohai_root": "/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/ohai-13.1.0/lib/ohai"
  },
  "chef": {
    "version": "13.1.31",
    "chef_root": "/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.1.31/lib"
  }
}
```

**Comment by lamont-granquist** [comment]
ultimate root cause seems to be that we're shipping two version of chef-config and the ohai binstub isn't pinning the chef gem and the ohai binstub is picking up the wrong chef-config version:

```
% cat /usr/local/bin/chef-client
[...]
gem "chef-config", "= 14.0.202"
[...]
gem "ohai", "= 14.0.29"
[...]
gem "chef", "= 14.0.202"
[...]

% cat /usr/local/bin/ohai
[...]
gem "chef-config", "= 14.0.190"
[...]
gem "ohai", "= 14.0.31"

% /opt/chef/embedded/bin/gem list | egrep chef
chef (14.0.202)
chef-config (14.0.202, 14.0.190)
```

there's a simple solution of tracking down why there's two chef-configs and removing that duplication.

another possible solution would be to going to the chef-dk style of appbundling where all the pins from the entire Gemfile.lock are applied to all the binstubs, which would prevent the entire class of lazy-loading gems that conflict, but would likely require fixing the issue in that style of using appbundler where it cannot pull from git sources (which should really get fixed so that we can pull from git sources in chef-dk as well).

**Comment by lamont-granquist** [comment]
It works in ChefDK:

```
# /opt/chefdk/bin/ohai chef_packages
[2018-04-26T14:26:20-07:00] INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping...
{
  "chef": {
    "version": "13.8.5",
    "chef_root": "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.8.5/lib"
  },
  "ohai": {
    "version": "13.8.0",
    "ohai_root": "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/ohai-13.8.0/lib/ohai"
  }
}
```

so the chef-dk style of appbundling chef+ohai does fix this problem.

**Comment by lamont-granquist** [comment]
PR that was trying to address this that got stalled:

https://github.com/chef/omnibus-software/pull/950

**Comment by lamont-granquist** [comment]
This got fixed in Chef 15.0:

```
#  /opt/chef/bin/ohai chef_packages
{
  "chef": {
    "version": "15.1.36",
    "chef_root": "/opt/chef/embedded/lib/ruby/gems/2.6.0/gems/chef-15.1.36/lib"
  },
  "ohai": {
    "version": "15.1.5",
    "ohai_root": "/opt/chef/embedded/lib/ruby/gems/2.6.0/gems/ohai-15.1.5/lib/ohai"
  }
}
```
